### DISCUSSION TITLE
The identifier 'Binary:Wix4UtilCA_X86' could not be found

### DISCUSSION THREAD

**Discussion by guenhter** [wixtoolset/issues#8581] [Questions]
Hi,

I used this tutorial https://wixtoolset.org/docs/tools/wixext/quietexec/ and created a wxs file containing exaclty the block of the instructions:

```xml
<SetProperty
  Id="WixQuietExecCmdLine"
  Value="&quot;[%ComSpec]&quot; /c ECHO %NUMBER_OF_PROCESSORS%"
  Before="ConvolutedCoreCountCustomAction"
  Sequence="execute"
  />
<CustomAction
  Id="ConvolutedCoreCountCustomAction"
  BinaryRef="Wix4UtilCA_$(sys.BUILDARCHSHORT)"
  DllEntry="WixQuietExec"
  Execute="immediate"
  Return="check"
  />

<InstallExecuteSequence>
  <Custom Action="ConvolutedCoreCountCustomAction" Before="AppSearch" />
</InstallExecuteSequence>
```

but when building the msi, the following error appears

```cmd
> wix build .\package.wxs -o my.msi
C:\Users\...\package.wxs(25) : error WIX0094: The identifier 'Binary:Wix4UtilCA_X86' could not be found. Ensure you have typed the reference correctly and that all the necessary inputs are provided to the linker.
```

---

I've installed Wix via `dotnet tool install --global wix`

```cmd
> wix --version
5.0.0+41e11442
```

any idea what I'm doing wrong?

**Comment by wmanning** [comment]
I think you need the util nuget package.  You didn't say what wix version but here's one for Wix5: https://www.nuget.org/packages/WixToolset.Util.wixext

**Reply by guenhter** [reply]
I'm using 

```
> wix --version
5.0.0+41e11442
```

but not a .net project. I'm creating Wix without any msbuild or thelike but just via cli. When I run the command:

```powershell
> dotnet add package WixToolset.Util.wixext --version 5.0.0
No project found in "C:\foo\".
```

**Reply by guenhter** [reply]
Maybe as additional background: Ultimatively I just wanna call a external scrpt to run this single line:

```batch
schtasks /create /f /tn "%TaskName%" /tr "%BatchFilePath%" /sc hourly
```

**Reply by wmanning** [reply]
Hmm, there's an artifacts zip on the wixtoolset release section containing all the nugets.  Nugets packages can be open like zip files and contains the dlls.  Guess you could get it from there.

I only used MSBuild and I don't know of a way to download nuget package without a project.

**Comment by robmen** [comment]
Read the parent topic in the documentation: [WiX extensions and custom actions](https://wixtoolset.org/docs/tools/wixext/)

**Reply by guenhter** [reply]
Thanks. This helped.

Is it possible to have a custom action then for installing and another one for deinstalling?
